File: python/06_071.py
def bucket_sort(arr, k):
    # Create k+1 buckets (for numbers 0 to k)
    count = [0] * (k + 1)
    
    # Count occurrences of each element
    for num in arr:
        count[num] += 1
    
    # Reconstruct the sorted array
    sorted_arr = []
    for i in range(k + 1):
        sorted_arr.extend([i] * count[i])
    
    return sorted_arr

File: data structures and algorithms/java/06_071.java
public static int[] bucketSort(int[] arr, int k) {
   // Create k+1 buckets (for numbers 0 to k)
   int[] count = new int[k + 1];
   
   // Count occurrences of each element
   for (int num : arr) {
       count[num]++;
   }
   
   // Reconstruct the sorted array
   int[] sortedArr = new int[arr.length];
   int index = 0;
   for (int i = 0; i <= k; i++) {
       for (int j = 0; j < count[i]; j++) {
           sortedArr[index++] = i;
       }
   }
   
   return sortedArr;
}